File: fabrevealmenu/src/main/java/com/hlab/fabrevealmenu/enums/Direction.java
package com.hlab.fabrevealmenu.enums;

public enum Direction {
    LEFT(0), UP(1), RIGHT(2), DOWN(3);
    private int id;

    Direction(int id) {
        this.id = id;
    }

    public static Direction fromId(int id) {
        for (Direction f : values()) {
            if (f.id == id) return f;
        }
        throw new IllegalArgumentException();
    }
}

File: fabrevealmenu/src/main/java/com/hlab/fabrevealmenu/enums/Direction.kt
package com.hlab.fabrevealmenu.enums

enum class Direction(private val id: Int) {
    LEFT(0), UP(1), RIGHT(2), DOWN(3);

    companion object {
        fun fromId(id: Int): Direction {
            for (f in values()) {
                if (f.id == id) return f
            }
            throw IllegalArgumentException()
        }
    }
}
